src/runtime_src/core/common/api/xrt_bo.cpp

```diff
@@ -368,8 +416,29 @@ class bo_impl
   {
     device->sync_aie_bo(bo, port.c_str(), dir, sz, offset);
   }
+
+  xrt::bo::async_handle
+  async(xrt::bo& bo, const std::string& port, xclBOSyncDirection dir, size_t sz, size_t offset)
+  {
+    device->sync_aie_bo_nb(bo, port.c_str(), dir, sz, offset);
+    auto a_bo_impl = std::make_shared<xrt::aie::bo::async_handle_impl>(bo, 0, port);
+    std::lock_guard lk(::async_bo_hdls_mutex);
+    ::async_bo_hdls[port].emplace_back(a_bo_impl.get());
```

Comment:
See also comment in wait().  I am thinking this doesn't belong here.  Abstract the map and management of the map into a class.   emplace as part of the contructor for xrt::aie::bo::async_handle_impl.

Reply:
Done